pennylane/qchem/structure.py

```diff
@@ -412,3 +413,136 @@ def excitations_to_wires(singles, doubles, wires=None):
         doubles_wires.append([d1_wires, d2_wires])
 
     return singles_wires, doubles_wires
+
+
+def pubchem_data(identifier, identifier_type="name"):
+    r"""Obtain name, geometry and charge of a compound from the PubChem Database.
+
+    The `PubChem <https://pubchem.ncbi.nlm.nih.gov>`__ is one of the largest public repositories for
+    the information on chemical substances from which name, geometry (in Bohr radius), and charge can be
+    retrieved for a compound by its name, SMILES, InChI, InChIKey, and PubChem Compound ID (CID) to
+    build a molecule object for Hartree-Fock calculations.
+
+    Args:
+        identifier (str or int): compound's identifier as required by the PubChem database
+        identifier_type (str): type of the provided identifier - name, CAS, CID, SMILES, InChI, InChIKey
+
+    Returns:
+        Tuple(list[str], array[float], int): symbols, geometry (in Bohr radius) and charge of the compound
+
+    **Example**
+
+    >>> mol_data_from_pubchem("BeH2")
+    (['Be', 'H', 'H'],
+     array([[ 4.79405604,  0.29290815,  0.        ],
+            [ 3.77946   , -0.29290815,  0.        ],
+            [ 5.80884105, -0.29290815,  0.        ]]),
+     0)
+    >>> mol_data_from_pubchem(223, "CID")
+    (['N', 'H', 'H', 'H', 'H'],
+    array([[ 4.79404621,  0.        ,  0.        ],
+            [ 5.80882913,  0.5858151 ,  0.        ],
+            [ 3.77945225, -0.5858151 ,  0.        ],
+            [ 4.20823111,  1.01459396,  0.        ],
+            [ 5.3798613 , -1.01459396,  0.        ]]),
+    1)
+
+    .. details::
+
+        **Usage Details**
+
+        ``mol_data_from_pubchem`` can also be used with other chemical identifiers - CAS, SMILES, InChI, InChIKey:
```

Comment:
Do we care about this? I have a hard time imaging that there is a user out there that would specify a molecule from CAS, SMILES, or InChIKey